xls/dslx/type_system/deduce_expr.cc

```diff
@@ -139,6 +139,15 @@ absl::StatusOr<std::unique_ptr<Type>> DeduceArray(const Array* node,
       member_types[0]->CloneToUnique(), member_types_dim);
 
   if (node->type_annotation() == nullptr) {
+    if (node->has_ellipsis()) {
+      const Span array_obrack_span(node->span().start(), node->span().start());
+      return TypeInferenceErrorStatus(
+        array_obrack_span, inferred.get(),
+        "Array has ellipsis (`...`) but does not have a type annotation; "
+        "please add a type annotation to indicate how many elements to expand to.",
```

Comment:
"For example: `u8[32]:[u8:0, ...]`"

Reply:
Good idea -- tried to make it relevant to what the user typed in instead of having a universal sample (which might be confusing if they didn't type in e.g. `u8` as the element type).